tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+        increment by one to add one frequency bin ON BOTH SIDES
+    noise_skip_neighborfreqs - int
+        set this >=1 if you want to exclude the immediately neighboring
+        frequency bins in noise level calculation
+
+    Returns
+    -------
+    snr - np.array
+        array containing snr for all epochs, channels, frequency bins.
+        NaN for frequencies on the edge, that do not have enoug neighbors on
+        one side to calculate snr
+    """
+
+    # prep not epoched / single channel data
+    is_2d = True if (len(psds.shape) == 2) else False
+    if is_2d:
+        psds = psds.reshape((1, psds.shape[0], psds.shape[1]))
+
+    # SNR loop
+    snr = np.empty(psds.shape)
+    for i_freq in range(psds.shape[2]):
+
+        # skip freqs on the edges (without noise neighbors)
+
+        start_freq_i = noise_n_neighborfreqs + noise_skip_neighborfreqs
+        stop_freq_i = (psds.shape[2] - noise_n_neighborfreqs
+                       - noise_skip_neighborfreqs)
+        if not (stop_freq_i > i_freq >= start_freq_i):
+            snr[:, :, i_freq] = np.nan
+            continue
+
+        # extract signal level
+        signal = psds[:, :, i_freq]
+
+        # ... and average noise level
+        i_noise = []
+        for i in range(noise_n_neighborfreqs):
+            i_noise.append(i_freq + noise_skip_neighborfreqs + i + 1)
+            i_noise.append(i_freq - noise_skip_neighborfreqs - i - 1)
+        noise = psds[:, :, i_noise].mean(axis=2)
+
+        snr[:, :, i_freq] = signal / noise
+
+    # reshape not epoched / single channel data to original dimensions
+    if is_2d:
+        snr = snr.reshape(snr.shape[1], snr.shape[2])
+
+    return snr
+
+
+###############################################################################
+# Calculate SNR
+# ^^^^^^^^^^^^^
+
+stim_freq = 12.
+noise_n_neighborfreqs = 3
+noise_skip_neighborfreqs = 1
+snrs = snr_spectrum(psds, noise_n_neighborfreqs=noise_n_neighborfreqs,
+                    noise_skip_neighborfreqs=noise_skip_neighborfreqs)
+
+###############################################################################
+# Find frequency bin containing stimulation frequency
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Ideally, this bin should should have the stimulation frequency exactly in the
+# center.
+
+tmp_distlist = abs(np.subtract(freqs, stim_freq))
+i_signal = np.where(tmp_distlist == min(tmp_distlist))[0][0]
+# could be updated to support multiple frequencies
+
+###############################################################################
+# Calculate SNR
+# ^^^^^^^^^^^^^
+# Extract and average SNRs at this frequency
+snrs_stim = snrs[:, :, i_signal]  # trial subselection can be done here
+print('average SNR at %iHz (all channels, all trials): %.3f '
+      % (stim_freq, snrs_stim.mean()))
+
+
+###############################################################################
+# Visualization
+# -------------
+
+
```

Comment:
i recently learned, one can apply seaborn style in pyplot by adding `plt.style.use('seaborn')` anywhere after the import statement

this would definitely lead to nicer figures! 
or are there any style guidelines from the mne side?

Reply:
I thought that seaborn style is applied automatically when you import seaborn and you don't have to explicitly activate it.

In any case, you do need to have seaborn installed for this to work which means that anyone trying to run our tutorial would have to install it. Which is probably not such a big deal. I've found one older mne-python issue where they decided against adding seaborn as a dependency. Two tutorials do already use seaborn which is why it is in the dependencies for docs. So, if we do import seaborn, the tutorials will be able to build without additional changes.

I suggest we try to import/activate seaborn before submitting PR upstream. If it makes a big difference - we add it If not - we don't.

Reply:
well, we dont import seaborn.

the code i proposed is just applies the style of seaborn (font, colors etc) to matplotlib.pyplot figures. it's implemented in matplotlib (> v.1.5) and doesnt need seaborn in the dependencies.

but youre right we can do this later